lib/client/client.go

```diff
@@ -108,20 +111,45 @@ func (c *Client) EnsureInitialized() error {
 		return err
 	}
 
-	// Ensure a cluster GUID is set.  This is required for all datastore
-	// types.
-	kv := &model.KVPair{
-		Key:   model.GlobalConfigKey{Name: "ClusterGUID"},
-		Value: hex.EncodeToString(uuid.NewV4().Bytes()),
+	// Ensure a number of global Felix config values are configured:
+	// -  GUID (ClusterGUID)
+	// -  IPIP (IpInIpEnabled)
+	// -  Felix reporting interval (ReportingIntervalSecs)
+	// -  Log severity for screen (LogSeverityScreen)
+	// -  Log severity for file (LogSeverityFile)
+	// -  Log file path (LogFilePath)
+	guid := hex.EncodeToString(uuid.NewV4().Bytes())
+	if err := c.ensureGlobalFelixConfig("ClusterGUID", guid); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("ReportingIntervalSecs", "0"); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("IpInIpEnabled", GlobalDefaultIPIP); err != nil {
+		return err
+	} else if err = c.ensureGlobalFelixConfig("LogSeverityScreen", GlobalDefaultLogLevel); err != nil {
```

Comment:
I agree wholeheartedly and I was in two minds about how to do this.

One reason i didn't do it in the node was because the current node and global config is configurable using set/unset/get commands there is no concept of assign a value if not already assigned - so perhaps what I should really be doing is updating the calico/node image to handle values not yet being assigned, and make sure they use the global default value.  That in itself a bit of a PITA, partly because it requires a bunch of rework to the templates, and partly because it means we have to store the "default" values in both the golang code and the templates.

But that said, maybe we should bit the bullet and update the templates to handle not having default values assigned.  WDYT?